### PR TITLE
NR-234826 - Missing DT Headers

### PR DESCRIPTION
**Pull request by smalsam-newr**
In several instances, Distributed Tracing headers were not added to requests before they were sent off. This was due to not calling the proper method which would, simultaneously generate the DT headers, and add them to the request. Later, the Distributed Tracing Payload would be associated with the request, allowing for round trip tracing. In addition, DT headers were not being added to Swift Async requests before they went off, due to the delegate method setState not giving us a proper updated state in the newState parameter.

### CODE REVIEW

**Comment by cdillard-NewRelic** [comment]
Should this point at the `develop` branch?

File: Agent/Instrumentation/NSURLSession/NRMAURLSessionTaskOverride.m [hunk 1]
```diff
@@ -126,7 +126,7 @@ void NRMAOverride__urlSessionTask_SetState(NSURLSessionTask* task, SEL _cmd, NSU
                 
                 NSURL *url = [currentRequest URL];
                 if (url != nil &&
-                    newState != NSURLSessionTaskStateRunning && task.state == NSURLSessionTaskStateRunning) {
```
Comment (cdillard-NewRelic) [inline Agent/Instrumentation/NSURLSession/NRMAURLSessionTaskOverride.m:129]: See slack convo, is this change really needed? The intent is to only call contents within if statement once per request.

**Review by cdillard-NewRelic — APPROVED** [review]

**Review by mbruin-NR — APPROVED** [review]
